pkgs/applications/virtualization/kata-containers/default.nix

```diff
@@ -0,0 +1,190 @@
+    "GOCACHE=$(TMPDIR)/go-cache"
+    "PREFIX=$(out)"
+  ];
+
+  # default meta
+  meta = with stdenv.lib; {
+    homepage = "https://katacontainers.io/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ offline ];
+    platforms = platforms.linux;
+  };
+
+  agent = stdenv.mkDerivation {
+    pname = "kata-agent";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "kata-containers";
+      repo = "agent";
+      rev = version;
+      sha256 = "6Zr7qwBuidJmKdZL2VGmRlKthWletdc0fRS5kA54CHM=";
+    };
+
+    goPackagePath = "github.com/kata-containers/agent";
+
+    inherit setSourceRoot;
+
+    nativeBuildInputs = [ pkgconfig go ];
+    buildInputs = [ libseccomp ];
+
+    makeFlags = makeFlags ++ [ "UNIT_DIR=$(out)/lib/systemd/system" ];
+
+    meta = meta // {
+      description = "Kata Containers version 1.x agent";
+    };
+  };
+
+  ksm-throttler = stdenv.mkDerivation {
+    pname = "kata-ksm-throttler";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "kata-containers";
+      repo = "ksm-throttler";
+      rev = version;
+      sha256 = "l+3pMjsfxU8170JKslhqv7PbaU3ROSQoscGQNbNh3+k=";
+    };
+
+    goPackagePath = "github.com/kata-containers/ksm-throttler";
+
+    nativeBuildInputs = [ go ];
+
+    inherit setSourceRoot makeFlags;
+
+    meta = meta // {
+      description = "Kata Containers KSM throttling daemon";
+    };
+  };
+
+  proxy = stdenv.mkDerivation {
+    pname = "kata-proxy";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "kata-containers";
+      repo = "proxy";
+      rev = version;
+      sha256 = "M2WaODisDPcpSc5AaMJEWI6MiF8Q1Iz/R6HzqFycOC8=";
+    };
+
+    goPackagePath = "github.com/kata-containers/proxy";
+
+    nativeBuildInputs = [ go ];
+
+    inherit setSourceRoot makeFlags;
+
+    meta = meta // {
+      description = "Kata Containers version 1.x proxy";
+    };
+  };
+
+  shim = stdenv.mkDerivation {
+    pname = "kata-shim";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "kata-containers";
+      repo = "shim";
+      rev = version;
+      sha256 = "sQHGCQ1Vowd6Zvm0wrN5Y2lEPzySkE1xlBt7tn3NppU=";
+    };
+
+    inherit setSourceRoot makeFlags;
+
+    goPackagePath = "github.com/kata-containers/shim";
+
+    nativeBuildInputs = [ go ];
+
+    meta = meta // {
+      description = "Kata Containers version 1.x shim";
+    };
+  };
+
+  runtime = stdenv.mkDerivation {
+    pname = "kata-runtime";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "kata-containers";
+      repo = "runtime";
+      rev = version;
+      sha256 = "TsXkkENZhH/p2SA4ytsWTiY5SJX+9FxUiX1o/v9RObo=";
+    };
+
+    goPackagePath = "github.com/kata-containers/runtime";
+
+    nativeBuildInputs = [ yq-go go ];
+
+    inherit setSourceRoot makeFlags;
+
+    patchPhase = ''
+      substituteInPlace Makefile --replace '-i -o' '-o'
+      substituteInPlace golang.mk --replace '$(GOPATH)/bin/yq' '${yq-go}/bin/yq'
```

Comment:
We might be able to drop this and `yq-go` by using `SKIP_GO_VERSION_CHECK`?

https://github.com/kata-containers/runtime/blob/master/Makefile#L15
https://github.com/kata-containers/runtime/blob/master/golang.mk#L17

Reply:
I can confirm this works, implemented suggested change.